lighthouse-core/audits/byte-efficiency/large-javascript-libraries.js

```diff
@@ -0,0 +1,160 @@
+
+const libStats = require('./bundlephobia-database.json');
+const librarySuggestions = require('./library-suggestions.json');
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when no known unnecessarily large libraries are detected on the page.*/
+  title: 'Avoids unnecessarily large JavaScript libraries',
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when some known unnecessarily large libraries are detected on the page.*/
+  failureTitle: 'Replace unnecessarily large JavaScript libraries',
+  /** Description of a Lighthouse audit that tells the user why they should care about the large Javascript libraries that have better alternatives.. This is displayed after a user expands the section to see more. No character length limits. */
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    ' [Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of large JS libraries that could be replaced. */
+  name: 'Library Name',
+  /** Label for a column in a data table. Entries will be names of smaller libraries that could be used as a replacement. */
+  suggestion: 'Smaller Alternative',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargeJavascriptLibraries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts.DetectedStack} library
+   * @return {string[]}
+   */
+  static getSuggestions(library) {
+    for (const key of Object.keys(librarySuggestions)) {
+      if (librarySuggestions[key].includes(library.npm)) {
+        return librarySuggestions[key].filter(suggestion => suggestion !== library.npm);
+      }
+    }
+    return [];
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: BundlePhobiaLibrary, suggestion: BundlePhobiaLibrary}>} */
+    const libraryPairings = [];
+    const foundLibraries = artifacts.Stacks.filter(stack => stack.detector === 'js');
```

Comment:
all stacks are detected by `js` atm.

It seems like more was planned for this artifact than ever materialized? https://github.com/GoogleChrome/lighthouse/pull/7243#issuecomment-467216423

in any case, the filter is fine i guess.